python/src/com/jetbrains/python/console/PyConsoleCopyHandler.kt

```diff
@@ -35,13 +35,13 @@ class PyConsoleCopyHandler(val originalHandler: EditorActionHandler) : EditorAct
 
   override fun doExecute(editor: Editor, caret: Caret?, dataContext: DataContext) {
     if (!RichCopySettings.getInstance().isEnabled) {
-      return originalHandler.execute(editor, null, dataContext);
+      return originalHandler.execute(editor, null, dataContext)
     }
-    if (true != editor.getUserData(ConsoleViewUtil.EDITOR_IS_CONSOLE_HISTORY_VIEW)) {
-      return originalHandler.execute(editor, null, dataContext);
-
+    if (true != editor.getUserData(ConsoleViewUtil.EDITOR_IS_CONSOLE_HISTORY_VIEW)
+        || editor.caretModel.allCarets.size != 1) {
```

Comment:
This fixes columnar selection. This basically skips this handler when in column selection mode. In row selection mode there is a single caret and in column selection mode there are multiple.